ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -221,7 +468,32 @@ let clean_local_resources () : unit =
     error "%s : %s" msg (ExnHelper.string_of_exn e) ;
     raise (Auth_service_error (E_GENERIC, msg))
 
+let domainify_uname ~domain uname =
+  let open Astring.String in
+  if
+    is_infix ~affix:domain uname
+    || is_infix ~affix:"@" uname
+    || is_infix ~affix:{|\|} uname
+    || uname = krbtgt
+  then
+    uname
+  else
+    Printf.sprintf "%s@%s" uname domain
+
 module AuthADWinbind : Auth_signature.AUTH_MODULE = struct
+  (* subject_id get_subject_identifier(string subject_name)
+
+      Takes a subject_name (as may be entered into the XenCenter UI when defining subjects --
+      see Access Control wiki page); and resolves it to a subject_id against the external
+      auth/directory service.
+      Raises Not_found (*Subject_cannot_be_resolved*) if authentication is not succesful.
+  *)
+  let get_subject_identifier subject_name =
+    let subject_name =
+      domainify_uname ~domain:(get_service_name ()) subject_name
```

Comment:
I believe it's a good idea to domainify the username here. I found instances where

```
wbinfo -n user@domain.net
```
works,

but
 
```
wbinfo -n user
```

does not work